### ISSUE TITLE
Migrate facility type

### ISSUE THREAD

**Issue by JaquM** [hzi-braunschweig/SORMAS-Project#4760]
### Feature Description
The structure of the stored facility types should be restructured due to the various new values, so that the corresponding stored facilities can also be found.

### Proposed Change

* Change the facility type group of the following facility types:

|Facility Type|Action
|-|-|
|HOSTEL|Migrate from RESIDENCE to ACCOMODATION|
|KINDERGARTEN| Migrate from CARE_FACILITY to EDUCATIONAL_FACILITY|

@kwa20 pls. check if: 
Database migration is (not) needed because facility type groups are not stored there.

### Additional Information

**Comment by bernardsilenou** [comment]
@JaquM 
- Both accommodation and residence has hotel as facility type, Think we only one, so we migrate, add to the other and delete the one under residence.
- I can remember that I asked the same question about point B and I was told that Kindergarten is not an educational facility according to the list that was provided, Has this changed?

**Comment by JaquM** [comment]
@bernardsilenou 
There is once hotel and once hostel, as I found it. Therefore both in accommodation. 
Kindergarten is more an educational than a care facility in Germany. That is why I would like to move this.

**Comment by MateStrysewske** [comment]
@bernardsilenou @JaquM Please let's use the categories that fit MOST of the time, not especially for Germany.

**Comment by JaquM** [comment]
@MateStrysewske 
If @bernardsilenou had the same idea, I would say it's for everybody the right way.

**Comment by bernardsilenou** [comment]
> @bernardsilenou
> There is once hotel and once hostel, as I found it. Therefore both in accommodation.
> Kindergarten is more an educational than a care facility in Germany. That is why I would like to move this.

 - Thanks, I think i need glasses :)
 - From international perspective, it is fine to have Kindergarten under educational facility. :+1:

**Comment by JaquM** [comment]
@MateStrysewske anything more to discuss?

**Comment by cazacmarin** [comment]
Findings:
- Was asked to change `Facility Category` for `Facility Type` HOSTEL and KINDERGARTEN.
- in DB this link is done via `healthfacility_id` column from `cases` table.
- If the modifications will be done only at code level, than older records will report to `Facility Category` and new records will report to changed  `Facility Category` 
- If consistency want to be kept, than a script which will modify all the cases from old `Facility Category`  to new one, should be implemented. (this script should modify the `healthfacility_id` column of existing cases with right IDs of updated `Facility Category`

**Comment by kwa20** [comment]
@cazacmarin The script for consistency sounds good. However, wouldn't we also have to update facility categories anywhere in the system and not just healthfacility_id?

**Comment by cazacmarin** [comment]
@kwa20 `healthfacility_id` is the linkage in DB in order to reach the  `facility categories`. So just changing `healthfacility_id`, will be enough. Old `facility categories` will go to the new one at the moment when this change will be performed.

**Comment by JaquM** [comment]
@cazacmarin is everything clear now with this?

**Comment by cazacmarin** [comment]
Hi @JaquM - yes, now all is clear. I'm removing `refine` label.

**Comment by cazacmarin** [comment]
For QA guys.
1) HOSTEL -> Migrate from RESIDENCE to ACCOMMODATION
Before merge of this branch into development:
Navigate to Cases directory and set `Facility category` to `Residence` and `Facility type` to `Hostel, dormitory`. As is the next image:
![image](https://user-images.githubusercontent.com/6985315/149714242-3adc4043-3611-40c7-887a-17192d0b748f.png)

After merge automatically  `Facility category` will change to:
![image](https://user-images.githubusercontent.com/6985315/149714292-a733eae3-12d3-4207-ab15-76f3111b8ad0.png)

2) KINDERGARTEN -> Migrate from CARE_FACILITY to EDUCATIONAL_FACILITY
Same for Immunization directory.
Before merge of this branch into development:
Navigate to Immunizations directory and set `Facility category` to `Care facility` and `Facility type` to `Kindergarten/After school care`. As is the next image:
![image](https://user-images.githubusercontent.com/6985315/149714878-31910d41-17dc-480e-a1ba-49df34dada0c.png)

After merge automatically  `Facility category` will change to:
![image](https://user-images.githubusercontent.com/6985315/149714936-de2345fe-938f-4abc-9a15-7bdc347ee305.png)


2nd Part. Those options will be moved to correspondent facility for new entity creation as well.

**Comment by mario-kuehne** [comment]
Succesfully tested locally with commit hash 71c2beefb2a9775997030f12202e3a6d99be801f.

**Comment by mario-kuehne** [comment]
Hint: change not yet deployed on any test instances. https://test.sormas.netzlink.com/ might be a good choice because immunization feature is enabled

**Comment by kwa20** [comment]
@mario-kuehne We also need to test the impact of this migration on existing data, especially in already existing facilities and existing references in associated entities. I'll create some test cases on [test-de2.sormas.netzlink.com](https://test-de2.sormas.netzlink.com/) for these scenarios

**Comment by kwa20** [comment]
@mario-kuehne Unfortunately, the instance restarted with the new SNAPSHOT version while I was creating the test data. However, I've managed to at least test Hostel: Residence > Accommodation. Looks good to me. The facility itself as well as the reference in the case (other entities not tested) were updated accordingly:

The facility was previously created with the category `RESIDENCE`

<img width="289" alt="facilityMigrationTestFacility" src="https://user-images.githubusercontent.com/70317594/149912516-c9660e0e-9741-4933-a8fd-380058d5da32.PNG">
<img width="396" alt="facilityMigrationTestCase" src="https://user-images.githubusercontent.com/70317594/149912518-35324ea2-3296-45ae-b99f-96682ed58bc8.PNG">
